app/translators/base.translator.js

```diff
@@ -0,0 +1,29 @@
+class BaseTranslator {
+  constructor (data) {
+    Object.assign(this, { _data: data })
+    // Set _data to be non-enumerable so it isn't passed to model validation
+    Object.defineProperty(this, '_data', { enumerable: false })
+
+    // Create each getter in _translations
+    // We do this within the constructor to keep Joi happy
+    for (const translator in this._translations) {
+      createStandardGetter(this, translator, this._translations[translator])
+    }
+  }
+
+  get _translations () {
+    throw new Error('You need to specify _translations in the child translator')
+  }
+}
+
+const createStandardGetter = (object, dataName, getterName) => {
```

Comment:
You might have just solved that problem we were looking at with the controllers (calling out to another method in the class from a route handler!)

However, in this context is there a reason I'm missing why `createStandardGetter` can't just be expressed as a simple method, for example, `_createStandardGetter(object, dataName, getterName)`?

Reply:
No reason! I'll change it to a function for clarity.